legend-engine-xts-data-space/legend-engine-xt-data-space-generation/src/main/java/org/finos/legend/engine/generation/analytics/DataSpaceAnalyticsHelper.java

```diff
@@ -398,9 +398,14 @@ else if (elementDoc instanceof Root_meta_pure_metamodel_dataSpace_analytics_Data
                     executableAnalysisResult.title = executable._title();
                     executableAnalysisResult.description = executable._description();
                     DataSpaceTemplateExecutableInfo templateExecutableInfo = new DataSpaceTemplateExecutableInfo();
+                    templateExecutableInfo.id = ((Root_meta_pure_metamodel_dataSpace_DataSpaceTemplateExecutable) executable)._id();
 
                     // get V1 lambda
-                    DataSpaceTemplateExecutable executableV1 = (DataSpaceTemplateExecutable) dataSpaceProtocol.executables.stream().filter(e ->  e instanceof DataSpaceTemplateExecutable && e.title.equals(executable._title()) && e.description.equals(executable._description())).findFirst().get();
+                    DataSpaceTemplateExecutable executableV1 = (DataSpaceTemplateExecutable) dataSpaceProtocol.executables.stream().filter(e ->
```

Comment:
we can just use the ID now to find the corresponding template executable now right?